### ISSUE TITLE
DWO, DWCO and CO should use CRD of the same versions

### ISSUE THREAD

**Issue by tolusha** [eclipse-che/che#19472]
### Is your task related to a problem? Please describe.
While working on https://github.com/eclipse/che/issues/19402 I faced a problem with starting DWCO in a separate container

```
2021-03-31T08:48:44.364Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "DevWorkspaceRouting.controller.devfile.io", "error": "no matches for kind \"DevWorkspaceRouting\" in version \"controller.devfile.io/v1alpha1\""}
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:143
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:184
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/manager/internal.go:661
2021-03-31T08:48:48.577Z	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "CheManager.che.eclipse.org", "error": "no matches for kind \"CheManager\" in version \"che.eclipse.org/v1alpha1\""}
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:143
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/internal/controller/controller.go:184
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.3/pkg/manager/internal.go:661
2021-03-31T08:48:48.577Z	ERROR	setup	problem running manager	{"error": "no matches for kind \"DevWorkspaceRouting\" in version \"controller.devfile.io/v1alpha1\""}
main.main
	/workspace/main.go:105
runtime.main
	/opt/rh/go-toolset-1.13/root/usr/lib/go-toolset-1.13-golang/src/runtime/proc.go:203
```
Basically it means that `DevWorkspaceRouting.controller.devfile.io` and `CheManager.che.eclipse.org` CRDs should exists on the cluster before starting container. That's not a problem at all because I can pre-create those CRD using `chectl` and include them into a operator bundle to deploy via olm. But leads to a problem that `che-operator` can't be deployed on OCP 4.4 and below (for OCP 3.11 we can do a hack) because DWO, DWCO use CRD v1 version which isn't supported by olm on those versions of OCP.

### Describe the solution you'd like
DWO, DWCO should use v1beta1 CRD version
Once we stop supporting 4.4 we can switch to CRD v1 [2]

### Additional context
[1] https://github.com/eclipse/che/issues/19145
[2] https://github.com/eclipse/che/issues/19460

/cc @l0rd @sleshchenko @metlos

**Comment by l0rd** [comment]
ack

**Comment by sleshchenko** [comment]
There is one issue with v1beta1 CRD:
> Warning: default unsupported in CRD version v1beta1, v1 required. Removing defaults.
And devworkspace uses it, search for `default: ` in https://github.com/devfile/api/blob/main/crds/workspace.devfile.io_devworkspaces.yaml
```
  sourceMapping:
    default: /projects
```
```
  exposure:
    default: public
```

Probably we should propagate the default behavior into DWO until we switch to v1 CRD API.

**Comment by amisevsk** [comment]
How long until 4.4 support is dropped? Since DevWorkspaces support is still in a quite early stage, I think it makes sense to restrict enabling DevWorkspace functionality to OpenShift 4.5 and greater. Otherwise we're potentially working around an issue for a short time, to support a minority of use cases.

**Comment by tolusha** [comment]
@amisevsk 

1. Right now we run che/crw operator on OCP 3.11 & 4.x
Since we are going to deploy DWCO in the same pod and support the same versions of OCP it requires crd v1beta1 version.

2. OCP 4.9 doesn't support crd v1beta1  it means we are obliged to switch to crd v1 [1] but we will still need to generate crd v1beta1 to support OCP 3.11.

[1] https://github.com/eclipse/che/issues/19460

**Comment by amisevsk** [comment]
> OCP 4.9 doesn't support crd v1beta1 it means we are obliged to switch to crd v1 [1] but we will still need to generate crd v1beta1 to support OCP 3.11.

DWO doesn't support 3.11, so this seems like a strange target.

**Comment by amisevsk** [comment]
To add: we'll continue to run into problems here. For example, conversion webhooks aren't supported until k8s 1.12, and OpenShift 3.11 uses k8s 1.11. 

Given that it'll be at least a few releases before DevWorkspace support is stable (and even more before it's a default), I don't think it makes sense to target releases of DWO on OpenShift 4.4, which is already in EOL.

**Comment by tolusha** [comment]
I see. It is harder to switch to CRD v1beta1 than expected.

Another way can be:
1. Postpone merging DWCO and CO [1] util OCP 4.4 is supported.
2. CO switches to CRD v1 (rollback DWCO to CRD v1)
3. Merge DWCO and CO 
4. CO uses a dedicated deployment file and  CRD v1beta1 to deploy on OCP 3.11

[1] https://github.com/eclipse/che/issues/19145

/cc @l0rd

**Comment by sleshchenko** [comment]
removing DWO label since we agreed that DWO will be on v1 CRD API

**Comment by tolusha** [comment]
Closed as WON'T FIX.
See comment https://github.com/eclipse/che/issues/19472#issuecomment-814655987
